pallets/utxo/src/staking.rs

```diff
@@ -30,8 +30,120 @@ use sp_runtime::transaction_validity::{TransactionLongevity, ValidTransaction};
 use sp_std::vec;
 
 use crate::staking::utils::remove_locked_utxos;
+use sp_runtime::DispatchError;
+use sp_std::marker::PhantomData;
 pub use validation::*;
 
+pub struct UtxoBalance<T>(PhantomData<T>);
+
+impl<T: Config> pallet_utxo_staking::Balance<T::AccountId> for UtxoBalance<T> {
+    fn staking_fee() -> Value {
+        T::StakeWithdrawalFee::get()
+    }
+
+    fn minimum_stake_balance() -> Value {
+        T::MinimumStake::get()
+    }
+
+    fn can_spend(account: &T::AccountId, value: Value) -> bool {
+        let (total, _, _) = pick_utxo::<T>(account, value);
+
+        total >= value
+    }
+
+    fn lock_for_staking(
+        stash: T::AccountId,
+        controller: T::AccountId,
+        session_keys: Vec<u8>,
+        value: Value,
+    ) -> DispatchResultWithPostInfo {
+        let (total, hashes, utxos) = pick_utxo::<T>(&stash, value);
```

Comment:
This is really bad from a practical point of view. So we're looping over all existing UTXOs to find some UTXOs for this user, so that we can lock for him. We're acting like a wallet... but not exactly, because we have to loop over everything.

The more practical way is to have the user create the transaction and spend the outputs he thinks are fit to be staked. Then, in the `spend` function, while verifying, we see what outputs he created, and the user chooses what outputs should be used for staking.